homeassistant/components/saj/sensor.py

```diff
@@ -185,15 +102,31 @@ def __init__(self, serialnumber, pysaj_sensor, inverter_name=None):
     @property
     def name(self):
         """Return the name of the sensor."""
-        if self._inverter_name:
-            return f"saj_{self._inverter_name}_{self._sensor.name}"
+        if self.coordinator.name != DEFAULT_NAME:
+            return f"{self.coordinator.name} {self._sensor.name}"
 
-        return f"saj_{self._sensor.name}"
+        return f"SAJ Inverter {self._sensor.name}"
 
     @property
     def native_value(self):
         """Return the state of the sensor."""
-        return self._state
+        return self._sensor.value
+
+    @property
+    def available(self) -> bool:
+        """
+        Return True if device is available.
+
+        Note: SAJ inverters are powered by DC via solar panels and thus are
+        offline after the sun has set, but we keep some sensor available.
```

Comment:
So, if I understand it correctly, it keeps the entity available, while the device is actually already offline?

Reply:
Yes, some sensor

Reply:
But that sounds incorrect, if a device is unavailable, the entities should reflect that.

Reply:
it is helpful to see the today_time, today_yield, total_co2_reduced, total_time, and total_yield even when the saj is offline, the current_power/temperature are valid when the saj is online

Reply:
> it is helpful to see the today_time, today_yield, total_co2_reduced, total_time, and total_yield even when the saj is offline, the current_power/temperature are valid when the saj is online

Instead, those sensors could be exempted from being unavailable maybe?

Reply:
This is what the code does. I don't get it. :)